rootfs/scheduler/resources/deployment.py

```diff
@@ -237,6 +237,12 @@ def in_progress(self, namespace, name, timeout, batches, replicas, tags):
             self.log(namespace, 'Deploy operation for Deployment {} in has expired. Rolling back to last good known release'.format(name), level='DEBUG')  # noqa
             return False, True
 
+        try:
+            self._check_for_failed_events(namespace, labels=labels)
```

Comment:
how come we need to check for failed events when determining that the deploy is in progress as well as when the replicas are ready?